apps/meteor/client/components/message/content/reactions/Reaction.tsx

```diff
@@ -41,29 +43,53 @@ const Reaction = ({ hasReacted, counter, name, names, ...props }: ReactionProps)
 	const ref = useRef<HTMLDivElement>(null);
 	const openTooltip = useTooltipOpen();
 	const closeTooltip = useTooltipClose();
+	const { showRealName } = useContext(MessageListContext);
 
 	const mine = hasReacted(name);
 
 	const key = getTranslationKey(names, mine);
 
 	const emojiProps = getEmojiClassNameAndDataTitle(name);
 
+	const getNames = useEndpoint('GET', '/v1/users.getNames');
+
+	const { refetch } = useQuery(
+		['users.getNames', names],
+		async () => {
+			if (names.length === 0) {
+				return undefined;
+			}
+
+			const users: string[] = showRealName
+				? (await getNames({ usernames: names }))?.users?.map((user) => user.name) || []
+				: names.map((name) => `@${name}`);
```

Comment:
about the `|| []` here, if there is a situation where the chain may break and result in undefined or something (as the typing says here), wdyt about falling back to showing usernames and not nothing?

Reply:
Makes sense, it shouldn't happen but usernames is better than nothing.